src/core/edgeNetwork/sendEdgeNetworkRequestFactory.js

```diff
@@ -12,7 +12,23 @@ governing permissions and limitations under the License.
 
 import { ID_THIRD_PARTY_DOMAIN } from "../../constants/domains";
 import apiVersion from "../../constants/apiVersion";
-import { createCallbackAggregator, noop, uuid } from "../../utils";
+import {
+  createCallbackAggregator,
+  noop,
+  uuid,
+  assign,
+  isNil
+} from "../../utils";
+
+const notNil = value => !isNil(value);
+
+const processReturnValue = values => {
+  if (isNil(values)) {
+    return {};
+  }
+
+  return values.filter(notNil).reduce((acc, item) => assign(acc, item), {});
```

Comment:
I'm not understanding why this is necessary. I ran this little prototype in my browser console:

![image](https://user-images.githubusercontent.com/210820/78694683-411ea900-78ba-11ea-9f97-e05247d60140.png)

which seems to suggest you could do this down below:

```
const lifecycleOnResponseReturnValues = segmentedReturnValues[0] || [];
const consumerOnResponseReturnValues = segmentedReturnValues[1] || [];
return assign({}, ...lifecycleOnResponseReturnValues, ...consumerOnResponseReturnValues);
```

Did you try something like that?